spec/requests/admin_amendments_spec.rb

```diff
@@ -0,0 +1,121 @@
+require "rails_helper"
+
+RSpec.describe "Admin claim amendments" do
+  let(:claim) { create(:claim, :submitted, teacher_reference_number: "1234567", bank_sort_code: "010203", date_of_birth: 25.years.ago.to_date) }
+
+  context "when signed in as a service operator" do
+    let(:service_operator) { create(:dfe_signin_user) }
+    before do
+      sign_in_to_admin_with_role(DfeSignIn::User::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE, service_operator.dfe_sign_in_id)
+    end
+
+    describe "admin_amendments#create" do
+      it "creates an amendment and updates the claim" do
+        old_date_of_birth = claim.date_of_birth
+        new_date_of_birth = 30.years.ago.to_date
+        expect {
+          post admin_claim_amendments_url(claim, amendment: {claim: {teacher_reference_number: "7654321", "bank_sort_code": "111213", "date_of_birth(3i)": new_date_of_birth.day, "date_of_birth(2i)": new_date_of_birth.month, "date_of_birth(1i)": new_date_of_birth.year},
+                                                             notes: "Claimant made a typo",})
+        }.to change { claim.reload.amendments.size }.by(1)
+
+        expect(response).to redirect_to(admin_claim_url(claim))
+
+        amendment = claim.amendments.last
+        expect(amendment.claim_changes).to eq({
+          "teacher_reference_number" => ["1234567", "7654321"],
+          "bank_sort_code" => ["010203", "111213"],
+          "date_of_birth" => [old_date_of_birth, new_date_of_birth],
+        })
+        expect(amendment.claim_changes).to be_a(ActiveSupport::HashWithIndifferentAccess)
+        expect(amendment.notes).to eq("Claimant made a typo")
+        expect(amendment.created_by).to eq(service_operator)
+
+        expect(claim.teacher_reference_number).to eq("7654321")
+        expect(claim.bank_sort_code).to eq("111213")
+        expect(claim.date_of_birth).to eq(new_date_of_birth)
+      end
+
+      it "saves the normalised value in the amendment when updating bank sort code" do
```

Comment:
I have some request specs here which I wrote before extracting some code from the controller into the model. So now there is some overlap between these specs and the model specs. I'm wondering whether this is a problem. In general, we don't seem to mind a bit of overlap between specs.

Reply:
👍